src/js/content.js

```diff
@@ -4,292 +4,317 @@ chrome.storage.local.get('tab_modifier', function (items) {
     if (items.tab_modifier === undefined) {
         return;
     }
-    
-    var tab_modifier = items.tab_modifier, rule = null, processPage;
-    
+
+    var processPage;
     processPage = function () {
+        var processRule;
```

Comment:
Rule processing moved into function that returns true if processed successfully and false otherwise.